pkgs/applications/audio/cheesecutter/default.nix

```diff
@@ -0,0 +1,48 @@
+{ stdenv, lib, fetchFromGitHub, fetchpatch
+, acme, ldc, makeWrapper
+, SDL
+}:
+
+stdenv.mkDerivation rec {
+  pname = "cheesecutter";
+  version = "unstable-2019-12-06";
+
+  src = fetchFromGitHub {
+    owner = "theyamo";
+    repo = "CheeseCutter";
+    rev = "6b433c5512d693262742a93c8bfdfb353d4be853";
+    sha256 = "1szlcg456b208w1237581sg21x69mqlh8cr6v8yvbhxdz9swxnwy";
+  };
+
+  buildInputs = [ SDL ];
+
+  nativeBuildInputs = [ ldc acme makeWrapper ];
+
+  patches = [
+    ./0001-fix-impure-build-date-display.patch
+  ];
+
+  makefile = "Makefile.ldc";
+
+  installFlags = [ "DESTDIR=$(out)" ];
+
+  postInstall = ''
+    cd $out
+    mkdir -p {bin,share/cheesecutter}
+    mv -t bin ccutter ct2util
+    mv -t share/cheesecutter example_tunes
```

Comment:
This works around the weird install target of the Makefile, which just dumps binaries & example files into $DESTDIR. I'll make that abit prettier in abit by just replacing the installPhase.